src/pages/smart-contracts/clarity-values-guide.md

```diff
@@ -0,0 +1,143 @@
+
+In order to build web applications that interact with Clarity contracts, we must include type information as a part of our function calls. The [@stacks/transactions](https://github.com/blockstack/stacks.js/tree/master/packages/transactions) library makes this easy, as we will demonstrate below.
+
+## Clarity Types
+The Clarity type system includes the following types:
+
+- `(tuple (key-name-0 key-type-0) (key-name-1 key-type-1) ...)` - a typed tuple with named fields.
+- `(list max-len entry-type)` - a list of maximum length max-len, with entries of type entry-type
+- `(response ok-type err-type)` - object used by public functions to commit their changes or abort. May be returned or used by other functions as well, however, only public functions have the commit/abort behavior.
+- `(optional some-type)` - an option type for objects that can either be (some value) or none
+- `(buff max-len)` - byte buffer or maximum length max-len.
+- `(string-ascii max-len)` - ASCII string of maximum length max-len
+- `(string-utf8 max-len)` - UTF-8 string of maximum length max-len
+- `principal` - object representing a principal (whether a contract principal or standard principal).
+- `bool` - boolean value (true or false)
+- `int` - signed 128-bit integer
+- `uint` - unsigned 128-bit integer
+
+## Constructing Clarity Values
+Clarity values can be constructed with functions provided by the **@stacks/transactions** library. These function simply output javascript objects that contain a value and a numerical representation of the Clarity type information (or in the case of Booleans and Optional None, just the type information). These Clarity value objects can then be easily serialized into `contract-call` transactions for interacting with Clarity contracts.
+
+Here are examples of how to construct each type of Clarity value:
+
+### Booleans
+```javascript
+const t = trueCV();
+const f = falseCV();
+```
+### Optional Values
+```javascript
+const nothing = noneCV();
+const something = someCV(t);
+```
+
+### Buffers
+```javascript
+const buffer = Buffer.from('foo');
+const bufCV = bufferCV(buffer);
+```
+
+### Integers
+```javascript
+const i = intCV(-10);
+const u = uintCV(10);
+```
+
+### Strings
+```javascript
+const ascii = stringAsciiCV('hello world');
+const utf8 = stringUtf8CV('hello 🌾');
+```
+
+### Principals
+```javascript
+const address = 'SP2JXKMSH007NPYAQHKJPQMAQYAD90NQGTVJVQ02B';
+const contractName = 'contract-name';
+
+const spCV = standardPrincipalCV(address);
+const cpCV = contractPrincipalCV(address, contractName);
+```
+
+### Response Values
+```javascript
+const errCV = responseErrorCV(trueCV());
```

Comment:
Is it correct to assume that folks can't know the response type ahead of the call completion? In that case, we should explain as well how the response can be identified (ie is this an error or OK response)?

Reply:
`Response` is a single type, with two possible values, `Ok(val)` and `Err(val)`.

Definitely safe and correct to assume that it won't be possible to anticipate if a successful response or an error response will be returned by a function that returns a `Response`.

I think I'll add a section below on how to detect what type a Clarity Value is